code/scripting/api/objs/ship.cpp

```diff
@@ -2337,6 +2338,147 @@ ADE_FUNC(setGlowPointBankActive, l_Ship, "boolean active, [number bank]", "Activ
 	return ADE_RETURN_NIL;
 }
 
+ADE_FUNC(setDocked, l_Ship, "ship dockee_ship, [string | number docker_point, string | number dockee_point]", "Immediately docks this ship with another ship.", nullptr, "Returns nothing")
+{
+	object_h *docker_objh = nullptr, *dockee_objh = nullptr;
+	int docker_point = -1, dockee_point = -1;
+
+	if (!ade_get_args(L, "oo", l_Ship.GetPtr(&docker_objh), l_Ship.GetPtr(&dockee_objh)))
+		return ADE_RETURN_NIL;
+
+	if (docker_objh == nullptr || dockee_objh == nullptr || !docker_objh->IsValid() || !dockee_objh->IsValid())
+		return ADE_RETURN_NIL;
+
+	auto docker_shipp = &Ships[docker_objh->objp->instance];
+	auto dockee_shipp = &Ships[dockee_objh->objp->instance];
+
+	auto docker_pm = model_get(Ship_info[docker_shipp->ship_info_index].model_num);
+	auto dockee_pm = model_get(Ship_info[dockee_shipp->ship_info_index].model_num);
+
+	if (lua_isnumber(L, 3))
+	{
+		ade_get_args(L, "**i", &docker_point);
+		docker_point--;	// Lua --> C/C++
+	}
+	else if (lua_isstring(L, 3))
+	{
+		const char *name = nullptr;
+		ade_get_args(L, "**s", &name);
+		docker_point = model_find_dock_name_index(Ship_info[docker_shipp->ship_info_index].model_num, name);
+	}
+	else
+	{
+		docker_point = 0;
+	}
+
+	if (lua_isnumber(L, 4))
+	{
+		ade_get_args(L, "***i", &dockee_point);
+		dockee_point--;	// Lua --> C/C++
+	}
+	else if (lua_isstring(L, 4))
+	{
+		const char* name = nullptr;
+		ade_get_args(L, "***s", &name);
+		dockee_point = model_find_dock_name_index(Ship_info[dockee_shipp->ship_info_index].model_num, name);
+	}
+	else
+	{
+		dockee_point = 0;
+	}
+
+	if (docker_point < 0 || docker_point >= docker_pm->n_docks)
+	{
+		Warning(LOCATION, "Invalid docker point specified for ship '%s'", docker_shipp->ship_name);
+		return ADE_RETURN_NIL;
+	}
+
+	if (dockee_point < 0 || dockee_point >= dockee_pm->n_docks)
+	{
+		Warning(LOCATION, "Invalid dockee point specified for ship '%s'", dockee_shipp->ship_name);
+		return ADE_RETURN_NIL;
+	}
+
+	//Make sure that the specified dockpoints are all free (if not, do nothing)
+	if (dock_find_object_at_dockpoint(docker_objh->objp, docker_point) != nullptr ||
+		dock_find_object_at_dockpoint(dockee_objh->objp, dockee_point) != nullptr)
+	{
+		nprintf(("scripting", "Cannot dock %s to %s because at least one of the specified dockpoints is occupied", docker_shipp->ship_name, dockee_shipp->ship_name));
+		return ADE_RETURN_NIL;
+	}
```

Comment:
Since this can be a somewhat common failure case, I'd suggest making the function return a bool.
If we have an invalid input, just return nil as you do, if the docking fails due to circumstances of the surrounding state, return false, and if it succeeds, return true

Reply:
That's a good idea.